sentry/src/event_aggregator.rs

```diff
@@ -125,25 +126,35 @@ impl EventAggregator {
 
         let has_access = check_access(
             &app.redis,
-            &session,
+            session,
             &app.config.ip_rate_limit,
             &record.channel,
             events,
         )
         .await;
         if let Err(e) = has_access {
-            return Err(ResponseError::BadRequest(e.to_string()));
+            let err = match e {
+                AccessError::OnlyCreatorCanCloseChannel | AccessError::ForbiddenReferrer => {
+                    ResponseError::Forbidden(e.to_string())
+                }
+                AccessError::RulesError(error) => ResponseError::TooManyRequests(error),
+                AccessError::UnAuthenticated => ResponseError::Unauthorized,
+                _ => ResponseError::BadRequest(e.to_string()),
+            };
+            return Err(err);
         }
 
         events
             .iter()
             .for_each(|ev| event_reducer::reduce(&record.channel, &mut record.aggregate, ev));
 
-        if ANALYTICS_RECORDER.is_some() {
+        // only time we don't have session is during
+        // an unauthenticated close event
+        if ANALYTICS_RECORDER.is_some() && session.is_some() {
```

Comment:
please `match`, we will remove the `expect` and it will look more clean.

Reply:
we can't use match has the match arms would be different